specs/driver_spec.rb

```diff
@@ -70,11 +82,176 @@
       average = @driver.average_rating
       average.must_be :>=, 1.0
       average.must_be :<=, 5.0
+      average.must_equal 4.0
     end
 
     it "returns zero if no trips" do
       driver = RideShare::Driver.new(id: 54, name: "Rogers Bartell IV", vin: "1C9EVBRM0YBC564DZ")
       driver.average_rating.must_equal 0
     end
+
+    it "does not include the nil rating of a trip in progress" do
+      in_progress_trip_data = {
+        id: 10,
+        driver: @driver,
+        passenger: RideShare::Passenger.new(id: 1, name: "Ada", phone: "412-432-7640"),
+        start_time: Time.now,
+        end_time: nil,
+        cost: nil,
+        rating: nil,
+      }
+
+      in_progress_trip = RideShare::Trip.new(in_progress_trip_data)
+
+      @driver.add_trip(in_progress_trip)
+
+      @driver.average_rating.must_equal 4.0
+
+    end
+  end
+
+  describe "total_revenue method" do
```

Comment:
Well done, for the `total_revenue` method it looks like you handled the edge cases well!  🥇